fraud/bad_encoding.go

```diff
@@ -173,3 +175,27 @@ func (p *BadEncodingProof) Validate(header *header.ExtendedHeader) error {
 
 	return nil
 }
+
+// OnBEFP listens to Bad Encoding Fraud Proof and stops services immediately if it is received.
+func OnBEFP(ctx context.Context, s Subscriber, stop func(context.Context) error) {
+	subscription, err := s.Subscribe(BadEncoding)
+	if err != nil {
+		log.Errorw("failed to subscribe to bad encoding fraud proof", "err", err)
```

Comment:
If this actually occurs, the node shouldn't start. We shouldn't have/allow nodes running on the network that aren't subscribed to BEFP.

Reply:
We should separate the subscription step from the listening for Proof step somehow

Reply:
We should keep in mind if Subscribe failed the we have to call `stop()`. So Subscribe+Listen should be somewhere in one place.
We can implement smth similar to `fraudSession` - a custom struct that will keep subscription + stop function and it will call stop once Fraud proof will be received.

cc @Wondertan , @liamsi

Reply:
If we have subscription construction happen separately, and for some reason, that fails, fx will fail out and not allow the node to start.

Reply:
Ok, just saw that @vgonkivs already implemented something similar

Reply:
ack